### PR TITLE
Add suport for cluster cloning

### PR DESCRIPTION
**Pull request by jamengual**
## what
* Adding cluster clone support

## why
* in aws provider 3.1.15 clone support was added

## references
* https://github.com/hashicorp/terraform-provider-aws/issues/5286

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -40,7 +40,7 @@ variable "cluster_size" {
 
 variable "snapshot_identifier" {
   type        = string
-  default     = ""
+  default     = null
```
Comment (jamengual) [inline variables.tf:43]: if snapshot_identifier is = "" this error will show up:
```

Error: ConflictsWith

  on ../terraform-aws-rds-cluster/main.tf line 47, in resource "aws_rds_cluster" "primary":
  47: resource "aws_rds_cluster" "primary" {

"restore_to_point_in_time": conflicts with snapshot_identifier
```

File: main.tf [hunk 2]
```diff
@@ -104,6 +104,15 @@ resource "aws_rds_cluster" "primary" {
     }
   }
 
+  dynamic "restore_to_point_in_time" {
+    for_each = var.clone_configuration
```
Comment (osterman) [inline main.tf:108]: Use upstream names to avoid confusion. In this "restore_to_point_in_time"

Comment (jamengual) [reply main.tf:108]: good point

**Review by Gowiem — APPROVED** [review]
LGTM

**Pull request merged**